### PR TITLE
Feature/swift pm

### PR DESCRIPTION
**Pull request by emawby**
This PR adds SwiftPM support to OneSignal.

- Added Package.swift that describes the OneSignal library package
- Updated build script to copy over public header file into it's own directory
- Deleted unneeded files

To test using XCode’s built in Swift PM support

1. Open XCode and navigate to your App’s project editor.
2. Select the Swift Packages tab and click the “+” to add a new package
3. Paste in the OneSignal iOS SDK Github URL. **To test pre-merge use https://github.com/emawby/OneSignal-iOS-SDK.git**
4. Choose to version the package off of the feature/SwiftPM branch
5. Navigate to the OneSignalNotificationExtension Target -> Build Phases Tab. Under the “Link Binary with Libaries section clickthe “+” and add the OneSignal library.

Closes https://github.com/OneSignal/OneSignal-iOS-SDK/issues/179

---
This change is [Reviewable](https://reviewable.io/reviews/onesignal/onesignal-ios-sdk/712)

### CODE REVIEW

**Comment by emawby** [comment]
The unit test succeed locally

File: iOS_SDK/OneSignalSDK/SwiftPM/Public/Headers/OneSignal.h [hunk 1]
```diff
@@ -0,0 +1,535 @@
+/**
```
Comment (jkasten2) [inline iOS_SDK/OneSignalSDK/SwiftPM/Public/Headers/OneSignal.h:1]: If we nest this file under `OneSignal` in `Headers` then the same Objective-C import statement (`#import <OneSignal/OneSignal.h>`) works like we have for our frameworks today. 
`iOS_SDK/OneSignalSDK/SwiftPM/Public/Headers/OneSignal/OneSignal.h`

I tested moving this file locally and it seems to work for Objective-C, and without causing any issue with Swift.

Comment (emawby) [reply iOS_SDK/OneSignalSDK/SwiftPM/Public/Headers/OneSignal.h]: made the change!

File: iOS_SDK/OneSignalSDK/build_fat_framework.sh [hunk 2]
```diff
@@ -79,6 +79,11 @@ RELEASE_OUTPUT_FRAMEWORK_DIR="${WORKING_DIR}/Framework/OneSignal.framework"
 rm -rf "${WORKING_DIR}/Framework/OneSignal.framework"
 cp -a "${FINAL_FRAMEWORK}" "${WORKING_DIR}/Framework/OneSignal.framework"
 
+echo "Copying public header for SwiftPM"
+#Copy the public header to the SwiftPM public headers directory
+rm -rf "${WORKING_DIR}/SwiftPM/Public/Headers/OneSignal.h"
```
Comment (jkasten2) [inline iOS_SDK/OneSignalSDK/build_fat_framework.sh:84]: Let's remove the whole SwiftPM folder here.
```sh
rm -rf "${WORKING_DIR}/SwiftPM"
```

That way if we remove or move files around in this directory in future builds we don't have old file carry over.

Comment (emawby) [reply iOS_SDK/OneSignalSDK/build_fat_framework.sh]: fixed!

File: iOS_SDK/OneSignalSDK/SwiftPM/Public/Headers/OneSignal.h (hunk 1 above)
Comment (jkasten2) [reply iOS_SDK/OneSignalSDK/SwiftPM/Public/Headers/OneSignal.h]: 👍

File: iOS_SDK/OneSignalSDK/build_fat_framework.sh (hunk 2 above)
Comment (jkasten2) [reply iOS_SDK/OneSignalSDK/build_fat_framework.sh]: 👍

**Pull request merged**